servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/ObservabilityProvider.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.netty.internal;
+
+import io.servicetalk.transport.api.ConnectionObserver;
+import io.servicetalk.transport.api.ConnectionObserver.SecurityHandshakeObserver;
+import io.servicetalk.transport.api.TransportObserver;
+
+import javax.annotation.Nullable;
+
+/**
+ * A helper class which allows sharing observers between different layers and handlers.
+ */
+public final class ObservabilityProvider {
```

Comment:
I do not quite understand the reasoning behind this class. `connectionObserver` and `handshakeObserver` references are temporary, i.e. they can only be retrieved once which makes the contract very error prone and not very different than the connection attribute approach. My hope was that we can be explicit in contract while passing these objects between classes, can we try doing that?

Reply:
Updated according to the approach we discussed offline. Looks a lot better now!
I will kill `SECURITY_HANDSHAKE_OBSERVER` in a follow-up PR.
Will keep `ChannelCloseUtils` here, because it's related to the `ConnectionObserver` and there was a bug in tests that is fixed now.